syntax = "proto3";

message GateActionZoom {
    repeated string identifiers = 1;
}

message GateActionSelect {
    repeated string identifiers = 1;
}

message GateActionIsolate {
    repeated string identifiers = 1;
}

message CreateModule {
    string name = 1;
    string parent_identifier = 2;
    repeated string identifiers = 3;
}

message DeleteModule {
    string identifier = 1;
}

message Message {
    uint64 sequence_number = 1;
    string application = 2;
    oneof payload {
        GateActionZoom gate_action_zoom = 10;
        GateActionSelect gate_action_select = 11;
        GateActionIsolate gate_action_isolate = 12;
        CreateModule create_module = 13;
        DeleteModule delete_module = 14;
    }
}
